src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1481,6 +1489,42 @@ private AuthenticationResult localFlow(CallbackHandler callbackHandle,
 
         return null;
     }
+    
+    /**
+     * Add log event for request failure with a cause. 
+     */
+    private void addLogEventForRequestFailureWithCause(final String eventName, final AuthenticationRequest request, 
+            final AuthenticationException authenticationException) {
+        ClientAnalytics.logEvent(
```

Comment:
I simply want to make the event logging with failure cause more easy to use, which could be used in multiple places. Instead of calling needs to do new InstrumentationPropertiesBuilder(param, param).build() every single time, simply pass in all the required params, and leave all the details inside the method.